lisa/util/parallel.py

```diff
@@ -51,6 +88,7 @@ def wait_worker(self) -> bool:
             self._futures.remove(future)
             # exception will throw at this point
             self._callback(result)
+            self._future_task_map[future].close()
```

Comment:
If the close() is not necessary, the map is also not necessary too.

Reply:
The closed one may need to be removed from the map. It may create a reference, so some resource are not recycled.